File: Source/MKL/examples/vmlf/source/vdsincos.f
	programm MKL_VML_TEST

	include "func_interv.fi"

	real*8 dA(VEC_LEN)
	real*8 dB1(VEC_LEN)
	real*8 dB2(VEC_LEN)
	real*8 dB3(VEC_LEN)
	real*8 dB4(VEC_LEN)
	real*8 CurRMS1,MaxRMS1
	real*8 CurRMS2,MaxRMS2
	real*8 MaxRMS
	integer i

	MaxRMS=0.0
	do i=1,VEC_LEN
		dA(i)=VML_DSINCOS_BEG+((VML_DSINCOS_END-VML_DSINCOS_BEG)*i)/VEC_LEN
		dB1(i)=0;
		dB2(i)=0;
		dB3(i)=sin(dA(i))
		dB4(i)=cos(dA(i))
	end do

	call VDSINCOS(VEC_LEN,dA,dB1,dB2)

	print *,"vdSinCos test/example programm"
	print *,""
	print *,""
	print *,"           Argument                     vdSinCos                      SinCos"
	print *,"=============================================================================="
	do i=1,VEC_LEN
		print 10,dA(i),dB1(i),dB2(i),dB3(i),dB4(i)
		CurRMS1=(dB1(i)-dB3(i))/(0.5*(dB1(i)+dB3(i)))
		CurRMS2=(dB2(i)-dB4(i))/(0.5*(dB2(i)+dB4(i)))
		if(MaxRMS1<CurRMS1) MaxRMS1=CurRMS1
		if(MaxRMS2<CurRMS2) MaxRMS2=CurRMS2
	end do

	if(MaxRMS2<MaxRMS1) MaxRMS=MaxRMS1
	if(MaxRMS2>MaxRMS1) MaxRMS=MaxRMS2
	print *,""
	print 11,"Maximum relative error: ",MaxRMS

10	format(F22.14,F20.6,F10.6,F13.6,F10.6)
11	format(A,F25.14)

	end

File: Source/MKL/examples/vmlc/source/vdsincos.c
#include <stdio.h>
#include <math.h>
#include "func_interv.h"

#include "mkl_vml.h"

int main()
{
  double dA1[VEC_LEN],dB1[VEC_LEN],dB2[VEC_LEN],dB3[VEC_LEN],dB4[VEC_LEN];

  int i=0,vec_len=VEC_LEN;
  double CurRMS,MaxRMSsin=0,MaxRMScos=0;

  for(i=0;i<vec_len;i++) {
    dA1[i]=(double)(__VML_DSINCOS_BEG+((__VML_DSINCOS_END-__VML_DSINCOS_BEG)*i)/vec_len);
    dB1[i]=0;
    dB2[i]=0;
    dB3[i]=sin(dA1[i]);
    dB4[i]=cos(dA1[i]);
  }

#if defined(MKL_VML_MIXED)
  vdSinCos(vec_len,dA1,dB1,dB2);
#elif defined(MKL_VML_LOWER)
  vdsincos(&vec_len,dA1,dB1,dB2);
#elif defined(MKL_VML_UPPER)
  VDSINCOS(&vec_len,dA1,dB1,dB2);
#else
# error You must define MKL_VML_CDECL for CDECL call convention or MKL_VML_STDCALL for STDCALL call convention.
#endif


  printf("vdSinCos test/example programm\n\n");
  printf("        Argument         vdSinCos:   Sin        Cos           Sin        Cos\n");
  printf("===============================================================================\n");
  for(i=0;i<vec_len;i++) {
    printf("% 22.14f % 20.6f % 10.6f % 13.6f % 10.6f\n",dA1[i],dB1[i],dB2[i],dB3[i],dB4[i]);
    CurRMS=(dB1[i]-dB3[i])/(0.5*(dB1[i]+dB3[i]));
    if(MaxRMSsin<CurRMS) MaxRMSsin=CurRMS;
    CurRMS=(dB2[i]-dB4[i])/(0.5*(dB2[i]+dB4[i]));
    if(MaxRMScos<CurRMS) MaxRMScos=CurRMS;
  }
  printf("\n");
  printf("Maximum relative error (sin): %.2f\n",MaxRMSsin);
  printf("Maximum relative error (cos): %.2f\n",MaxRMScos);
}
